.pipelines/templates/aks-upgrade.yaml

```diff
@@ -0,0 +1,18 @@
+steps:
+  - script: |
+      AKS_UPGRADE_VERSION=$(az aks get-upgrades -g ${AZURE_CLUSTER_NAME} -n ${AZURE_CLUSTER_NAME} --query 'max(controlPlaneProfile.upgrades[].kubernetesVersion)' -otsv)
+      echo "##vso[task.setvariable variable=AKS_UPGRADE_VERSION]${AKS_UPGRADE_VERSION}"
+    displayName: "Get upgrade version"
+
+  - script: |
+      echo "Upgrading to Kubernetes ${AKS_UPGRADE_VERSION}"
+      az aks upgrade -g ${AZURE_CLUSTER_NAME} -n ${AZURE_CLUSTER_NAME} -k ${AKS_UPGRADE_VERSION} --yes
+
+      # Sleep for 120 seconds to wait for nodes and pods to become ready
+      sleep 2m
```

Comment:
is 2m enough?

Reply:
The 2m sleep is after upgrade to check health of the components. `az aks upgrade` is blocking until the upgrade is complete.